docs/runtime/instantiable-module.md

```diff
@@ -0,0 +1,140 @@
+
+
+## What is an Instantiable Module?
+An instantiable runtime module is, in most ways, just like any other runtime module. It defines storage items, events, dispatchable calls, and a genesis configuration. But with just a few extra bits of syntax, the module can be made instantiable which means it can be deployed multiple times in the same runtime.
+
+Some use cases:
+
+* Token chain hosts two independent cryptocurrencies.
+* Marketplace track users' reputations as buyers separately from their reputations as sellers.
+* Governance has two (or more) houses which act similarly internally.
+
+The SRML's Balances and Collective modules are good examples of real-world code using this technique.
+
+## Writing an Instantiable Module
+Writing an instantiable module is almost entirely the same process as writing a plain non-instantiable module. There are just a few places where the syntax differs. If you prefer video walkthroughs, you can see a [recording](https://www.youtube.com/watch?v=XEl59hVcyI8) of making these changes.
+
+### Configuration Trait
+```rust
+pub trait Trait<I: Instance>: system::Trait {
+	// TODO: Add other types and constants required configure this module.
+
+	/// The overarching event type.
+	type Event: From<Event<Self, I>> + Into<<Self as system::Trait>::Event>;
+}
+```
+
+### Storage Declaration
+```rust
+decl_storage! {
+	trait Store for Module<T: Trait<I>, I: Instance> as TemplateModule {
+		...
+	}
+}
+```
+
+### Declaring the Module Struct
+```rust
+decl_module! {
+	/// The module declaration.
+	pub struct Module<T: Trait<I>, I: Instance> for enum Call where origin: T::Origin {
+		...
+	}
+}
+```
+### Accessing Storage
+```rust
+<Something<T, I>>::put(something);
+```
+
+<!-- In v2.0 the T is omitted for both instantiable and non-instantiable modules
+<Something<I>>::put(something); -->
+
+### Event initialization
+```rust
+fn deposit_event<T, I>() = default;
+```
+
+### Event Declaration
+```rust
+decl_event!(
+	pub enum Event<T, I> where AccountId = <T as system::Trait>::AccountId {
+		...
+	}
+}
+```
+
+## Deploying an Instantiable Module
+
+When composing a runtime, one must take care to use the proper syntax for instantiable modules. As before it is slightly different.
+
+### Implementing the Configuration Trait
+When implementing the module's configuration trait, you need to be explicit which instance you are configuring, because in general the instances need not have identical configurations. In this snippet we've named the instance `Instance1`.
+```rust
+impl template::Trait<template::Instance1> for Runtime {
+	type Event = Event;
+}
+```
+
+### In the `construct_runtime!` Macro
+Finally, in the `construct_runtime!` macro, we give a more human-friendly name to each instance of the module. Here I've called `Instance1` `FirstTemplate`.
```

Comment:
"we give a more human-friendly name to each instance of the module"

This is absolutely not required when writing instantiate modules. You can do it in the examples for clarity, but I would omit this kind of text from the docs.

What you are doing when you add it to `construct_runtime` is that you are _actually_ adding the module to your runtime. If the module is not put here, it will not be "constructed" into the runtime.

This is why this step is important and what you should be telling the user.

Reply:
> This is absolutely not required when writing instantiate modules.

Yes, I agree, but

> If the module is not put here, it will not be "constructed" into the runtime.

gives the incorrect impression that removing the module from `construct_runtime!` is sufficient to remove it from the runtime. In fact making that change alone simply causes the runtime not to compile which is why I didn't use that phrasing.

I've edited in a way that I think addresses both concerns.